domains/zeto/internal/zeto/fungible/handler_transferLocked.go

```diff
@@ -0,0 +1,340 @@
+	if err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorFormatProvingReq, err)
+	}
+
+	return &pb.AssembleTransactionResponse{
+		AssemblyResult: pb.AssembleTransactionResponse_OK,
+		AssembledTransaction: &pb.AssembledTransaction{
+			InputStates:  inputStates,
+			OutputStates: outputStates,
+		},
+		AttestationPlan: []*pb.AttestationRequest{
+			{
+				Name:            "sender",
+				AttestationType: pb.AttestationType_SIGN,
+				Algorithm:       h.getAlgoZetoSnarkBJJ(),
+				VerifierType:    zetosignerapi.IDEN3_PUBKEY_BABYJUBJUB_COMPRESSED_0X,
+				PayloadType:     zetosignerapi.PAYLOAD_DOMAIN_ZETO_SNARK,
+				Payload:         payloadBytes,
+				Parties:         []string{tx.Transaction.From},
+			},
+		},
+	}, nil
+}
+
+func (h *transferLockedHandler) Endorse(ctx context.Context, tx *types.ParsedTransaction, req *pb.EndorseTransactionRequest) (*pb.EndorseTransactionResponse, error) {
+	return nil, nil
+}
+
+func (h *transferLockedHandler) Prepare(ctx context.Context, tx *types.ParsedTransaction, req *pb.PrepareTransactionRequest) (*pb.PrepareTransactionResponse, error) {
+	var proofRes corepb.ProvingResponse
+	result := domain.FindAttestation("sender", req.AttestationResult)
+	if result == nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorFindSenderAttestation)
+	}
+	if err := proto.Unmarshal(result.Payload, &proofRes); err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorUnmarshalProvingRes, err)
+	}
+
+	inputSize := common.GetInputSize(len(req.InputStates))
+	inputs, err := utxosFromInputStates(ctx, req.InputStates, inputSize)
+	if err != nil {
+		return nil, err
+	}
+	outputs, err := utxosFromOutputStates(ctx, req.OutputStates, inputSize)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := common.EncodeTransactionData(ctx, req.Transaction)
+	if err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorEncodeTxData, err)
+	}
+	params := map[string]any{
+		"inputs":  inputs,
+		"outputs": outputs,
+		"proof":   common.EncodeProof(proofRes.Proof),
+		"data":    data,
+	}
+	transferFunction := getTransferLockedABI(tx.DomainConfig.TokenName)
+	if common.IsNullifiersToken(tx.DomainConfig.TokenName) {
+		delete(params, "inputs")
+		params["nullifiers"] = strings.Split(proofRes.PublicInputs["nullifiers"], ",")
+		params["root"] = proofRes.PublicInputs["root"]
+	}
+	paramsJSON, err := json.Marshal(params)
+	if err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorMarshalPrepedParams, err)
+	}
+	functionJSON, err := json.Marshal(transferFunction)
+	if err != nil {
+		return nil, err
+	}
+
+	signer := &tx.Params.(*types.FungibleTransferLockedParams).Delegate
+	if req.Transaction.Intent == prototk.TransactionSpecification_PREPARE_TRANSACTION {
+		// if called as prepare, it means the delegate might not exist locally
+		// and the signer doesn't matter because the caller is interested in the
+		// prepared transaction, not the signature
+		signer = &req.Transaction.From
```

Comment:
Just another thing that feels very wonky about "prepare"

Reply:
"wonky" is the right word.